spring-boot-project/spring-boot/src/main/java/org/springframework/boot/cloud/CloudFoundryVcapEnvironmentPostProcessor.java

```diff
@@ -89,9 +90,9 @@
  * @author Andy Wilkinson
  * @since 1.3.0
  */
-public class CloudFoundryVcapEnvironmentPostProcessor implements EnvironmentPostProcessor, Ordered {
+public class CloudFoundryVcapEnvironmentPostProcessor implements EnvironmentPostProcessor, Ordered, ApplicationListener<ApplicationEvent> {
```

Comment:
You could implement `ApplicationListener<ApplicationPreparedEvent>` instead and avoid the if check in the listener implementation.

Reply:
Thank you for pointing that out. Fixed with https://github.com/spring-projects/spring-boot/pull/19027/commits/a9c1e031e8c64e7807964bc2c6f0fdc9e62515f8